app/src/main/java/fr/free/nrw/commons/nearby/NearbyMapFragment.java

```diff
@@ -73,6 +108,114 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         return mapView;
     }
 
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        this.getView().setFocusableInTouchMode(true);
+        this.getView().requestFocus();
+        this.getView().setOnKeyListener( new View.OnKeyListener() {
+
+            @Override
+            public boolean onKey( View v, int keyCode, KeyEvent event) {
+                if (keyCode == KeyEvent.KEYCODE_BACK) {
+                    if(bottomSheetDetailsBehavior.getState() == BottomSheetBehavior
+                            .STATE_EXPANDED) {
+                        bottomSheetDetailsBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
+                        return true;
+                    }
+                    else if (bottomSheetDetailsBehavior.getState() == BottomSheetBehavior
+                            .STATE_COLLAPSED) {
+                        bottomSheetDetailsBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);
+                        return true;
+                    }
+                }
+                return false;
+            }
+        } );
+    }
+
+    private void initViews() {
+        bottomSheetList = getActivity().findViewById(R.id.bottom_sheet);
+        bottomSheetListBehavior = BottomSheetBehavior.from(bottomSheetList);
+        bottomSheetDetails = getActivity().findViewById(R.id.bottom_sheet_details);
+        bottomSheetDetailsBehavior = BottomSheetBehavior.from(bottomSheetDetails);
+        bottomSheetDetailsBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);
+
+        fabList = getActivity().findViewById(R.id.fab_list);
+        fabPlus = getActivity().findViewById(R.id.fab_plus);
+        fabCamera = getActivity().findViewById(R.id.fab_camera);
+        fabGallery = getActivity().findViewById(R.id.fab_galery);
+
+        fab_open = AnimationUtils.loadAnimation(getActivity(), R.anim.fab_open);
+        fab_close = AnimationUtils.loadAnimation(getActivity(),R.anim.fab_close);
+        rotate_forward = AnimationUtils.loadAnimation(getActivity(),R.anim.rotate_forward);
+        rotate_backward = AnimationUtils.loadAnimation(getActivity(),R.anim.rotate_backward);
+
+        transparentView = getActivity().findViewById(R.id.transparentView);
+        moreInfo = getActivity().findViewById(R.id.more_info_button);
+
+        description = getActivity().findViewById(R.id.description);
+        title = getActivity().findViewById(R.id.title);
+        distance = getActivity().findViewById(R.id.category);
+        icon = getActivity().findViewById(R.id.icon);
+    }
+
+    private void setListeners() {
+        fabPlus.setOnClickListener(new View.OnClickListener() {
```

Comment:
These click listeners can also be replaced with Butterknife. If this proves to be too much of an effort we can probably just add a TODO and a new comer can pick it up. :)